aws/sdk/examples/lambda-invoke-function/src/main.rs

```diff
@@ -2,52 +2,139 @@
+    // 'Invoke' as part of the AWS Lambda API.
+    // https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html
+    //
+    // The Invoke API accepts several arguments, but we're only going to
+    // include the required arguments in this example. The function name can
+    // be the human friendly name with or without a version alias. It can be a
+    // full ARN, or a partial ARN that includes the AWS account ID. following 
+    // are examples of each that will point to the same AWS resource.
+    //
+    // short name: 'example-function'
+    // short name with version alias: 'example-function:LATEST
+    // partial ARN: 072326518754:function:example-function
+    // full ARN: arn:aws:lambda:us-west-2:072326518754:function:example-function
+    //
+    // We are going to use the full ARN to prevent any ambiguity in which
+    // function will be invoked.
     match client
         .invoke()
-        .function_name("arn:aws:lambda:us-west-2:892717189312:function:my-rusty-func")
+        .function_name("arn:aws:lambda:us-west-2:072326518754:function:hello-python")
         .send()
         .await
     {
+        // If the API call returns without an error, the Lambda Invoke API
+        // returns a response object containing the following:
+        //
+        // StatusCode: The HTTP status code is in the 200 range for a
+        //   successful request
+        //
+        // ExecutedVersion: The version of the function that executed. When
+        //   you invoke a function with an alias, this indicates which version
+        //   the alias resolved to.
+        //
+        // FunctionError: If present, indicates that an error occurred during
+        //   function execution. Details about the error are included in the
+        //   response payload.
+        //
+        // LogResult: The last 4 KB of the execution log, which is base64
+        //    encoded.
+        //
+        // Payload: The response from the function, or an error object.
+        //
+        // For our example, we are just going to interact with the payload
+        // portion of the response
         Ok(resp) => {
-            println!("Response:");
-            println!("  {:?}", resp.payload);
+            match resp.payload {
+                // I'm sure there is some better idiomatic Rust way to do this
+                // but here we are. The payload response is a blob of binary
+                // data that we need to turn into something useful.
+                Some(blob) => {
+                    // I assign the contents of the blob to another variable
+                    // so that I can borrow it later without the compiler being
+                    // mad at me.
+                    let random_inner_crap = blob.into_inner();
+                    let s = str::from_utf8(&random_inner_crap).unwrap();
+                    println!("Response:");
+                    println!("  {:?}", s);
```

Comment:
I am supersized that this
`println!("  {:?}", str::from_utf8(&blob.into_inner()).unwrap());`
gave you an error. What error did you get?

Reply:
I had just the second `let` with &blob.into_inner() and I was getting an error about the lifetime of the response from into_inner(). I didn't think to fold it all into a single print statement

Reply:
Can also do std::str::from_utf8(blob.as_ref())

A common Rust pattern would also be something like "let blob = blob.into_inner()" and shadow the old value.